doc/tutorials/content/grid_filters.rst

```diff
@@ -0,0 +1,69 @@
+.. _grid_filters:
+
+Design of VoxelGrid code structure
+----------------------------------
+
+This document is intended to explain our new ``VoxelGrid`` design and help you implementing your own grid based filters easily by reusing our implementations.
+
+Inheritance structure of VoxelGrid
+----------------------------------
+
+.. image:: images/voxel_grid_code_structure.jpg
```

Comment:
Would it be possible to use plantuml to generate the image and also add that code in the repo?

Reply:
https://github.com/PointCloudLibrary/pcl/blob/85fc1707dd28f4104a7cf6ab0f8570d11ec1b4c4/doc/doxygen/doxyfile.in#L358-L360

Do we have to set it up first?

Reply:
I actually don't know. I was referring to still adding a png/jpg, but adding a source code as well. If we can use plantuml in doxygen, that'd be crazy.

@larshg @SunBlack Any thoughts?

Reply:
In case it works, I think using the source code of an image is better, as we can than modify it later if necessary.